### DISCUSSION TITLE
Pulling my heart out with a type mismatch.

### DISCUSSION THREAD

**Discussion by NDR008** [trackmania-rl/tmrl#61] [Q&A]
step() and reset(() give me this warning:
```WARN: The obs returned by the `reset()` method was expecting numpy array dtype to be int64, actual type: int32```
Pulling my hair a bit on this one.

The observation looks "int64" to me (except maybe the action history at the end?)
```(array([1], dtype=int64), array([0], dtype=int64), array([600], dtype=int64), array([1], dtype=int64), array([0], dtype=int64), array([0], dtype=int64), array([0]), array([0]), array([0]), array([0]))```

**Comment by yannbouteiller** [comment]
Hi, I think you want to make sure that both the action returned by `get_default_action()` (defined in your `RealTimeGymInterface`) and the one that you send to `step` are numpy arrays of dtype `int64`

**Comment by NDR008** [comment]
I should have mentioned both step() and reset() throw the same warning

**Reply by yannbouteiller** [reply]
It is probably `get_default_action()` that returns an action of the wrong dtype then

**Comment by NDR008** [comment]
man... what did i break...
[image: image.png]
grrr
